core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -32,22 +32,59 @@
  */
 public abstract class RetryingClient<I extends Request, O extends Response>
         extends SimpleDecoratingClient<I, O> {
+
+    private static final int FALLBACK_DEFAULT_MAX_RETRY_ATTEMPTS = 5;
+    private static final int DEFAULT_MAX_RETRY_ATTEMPTS;
+
+    static {
+        String value = System.getProperty("com.linecorp.armeria.defaultMaxRetryAttempts", "5");
+        int maxRetryAttempts;
+        try {
+            maxRetryAttempts = Integer.parseInt(value);
+        } catch (Exception ignored) {
+            maxRetryAttempts = FALLBACK_DEFAULT_MAX_RETRY_ATTEMPTS;
+        }
+        DEFAULT_MAX_RETRY_ATTEMPTS = maxRetryAttempts;
+    }
+
     private final Supplier<? extends Backoff> backoffSupplier;
     private final RetryRequestStrategy<I, O> retryStrategy;
+    private int maxRetryAttempts;
 
     /**
      * Creates a new instance that decorates the specified {@link Client}.
      */
     protected RetryingClient(Client<? super I, ? extends O> delegate,
                              RetryRequestStrategy<I, O> retryStrategy,
                              Supplier<? extends Backoff> backoffSupplier) {
+        this(delegate, retryStrategy, backoffSupplier, DEFAULT_MAX_RETRY_ATTEMPTS);
+    }
+
+    /**
+     * Creates a new instance that decorates the specified {@link Client}.
+     */
+    protected RetryingClient(Client<? super I, ? extends O> delegate,
+                             RetryRequestStrategy<I, O> retryStrategy,
+                             Supplier<? extends Backoff> backoffSupplier,
+                             int maxRetryAttempts) {
```

Comment:
maxRetryAttempts -> defaultMaxAttempts

Please note that 'max attempts' includes the first attempt, which means maxAttempts is maxRetryAttempts + 1. It seems to me that it will be very confusing to mix the term retries and attempts here. How about using 'attempts' only?

Also, this parameter should be prefixed with 'default' because it's the default value which is used only when the backoff does not limit the number of attempts.